### PR TITLE
test: Migrate tests to Node test container

### PR DESCRIPTION
**Pull request by KristianLentino99**
Refactor tests to use `node:test` instead of `tap`
- Relevant to issue https://github.com/fastify/fastify/issues/5555
- Updated /test/async-dispose.test.js
- Updated /test/close-pipelining.test.js
- Updated /test/has-router.test.js

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by jsumners — APPROVED** [review]
Looks good to me.

File: test/async-dispose.test.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-t.test('async dispose should close fastify', { skip: !('asyncDispose' in Symbol) }, async t => {
+test('async dispose should close fastify', { skip: !('asyncDispose' in Symbol) }, async t => {
   t.plan(2)
 
   const fastify = Fastify()
 
   await fastify.listen({ port: 0 })
 
-  t.equal(fastify.server.listening, true)
+  t.assert.equal(fastify.server.listening, true)
```
Comment (gurgunday) [inline test/async-dispose.test.js:14]:
```suggestion
  t.assert.strictEqual(fastify.server.listening, true)
```

File: test/has-route.test.js [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
-  const test = t.test
-  const fastify = Fastify()
+const fastify = Fastify()
 
+describe('hasRoute', async t => {
   test('hasRoute - invalid options', t => {
     t.plan(3)
 
-    t.equal(fastify.hasRoute({ }), false)
+    t.assert.equal(fastify.hasRoute({ }), false)
```
Comment (dancastillo) [inline test/has-route.test.js:12]:
```suggestion
    t.assert.strictEqual(fastify.hasRoute({ }), false)
```

File: test/has-route.test.js [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
 
-    t.equal(fastify.hasRoute({ method: 'GET' }), false)
+    t.assert.equal(fastify.hasRoute({ method: 'GET' }), false)
```
Comment (dancastillo) [inline test/has-route.test.js:14]:
```suggestion
    t.assert.strictEqual(fastify.hasRoute({ method: 'GET' }), false)
```

File: test/has-route.test.js [hunk 4]
```diff
@@ -1,22 +1,19 @@
...
 
-    t.equal(fastify.hasRoute({ constraints: [] }), false)
+    t.assert.equal(fastify.hasRoute({ constraints: [] }), false)
```
Comment (dancastillo) [inline test/has-route.test.js:16]:
```suggestion
    t.assert.strictEqual(fastify.hasRoute({ constraints: [] }), false)
```

File: test/has-route.test.js [hunk 5]
```diff
@@ -29,12 +26,12 @@ test('hasRoute', t => {
       }
     })
 
-    t.equal(fastify.hasRoute({
+    t.assert.equal(fastify.hasRoute({
```
Comment (dancastillo) [inline test/has-route.test.js:29]:
```suggestion
    t.assert.strictEqual(fastify.hasRoute({
```

File: test/has-route.test.js [hunk 6]
```diff
@@ -29,12 +26,12 @@ test('hasRoute', t => {
...
       method: 'GET',
       url: '/'
     }), true)
 
-    t.equal(fastify.hasRoute({
+    t.assert.equal(fastify.hasRoute({
```
Comment (dancastillo) [inline test/has-route.test.js:34]:
```suggestion
    t.assert.strictEqual(fastify.hasRoute({
```

File: test/has-route.test.js [hunk 7]
```diff
@@ -51,13 +48,13 @@ test('hasRoute', t => {
       }
     })
 
-    t.equal(fastify.hasRoute({
+    t.assert.equal(fastify.hasRoute({
```
Comment (dancastillo) [inline test/has-route.test.js:51]:
```suggestion
    t.assert.strictEqual(fastify.hasRoute({
```

File: test/has-route.test.js [hunk 8]
```diff
@@ -51,13 +48,13 @@ test('hasRoute', t => {
...
       method: 'GET',
       url: '/',
       constraints: { version: '1.2.0' }
     }), true)
 
-    t.equal(fastify.hasRoute({
+    t.assert.equal(fastify.hasRoute({
```
Comment (dancastillo) [inline test/has-route.test.js:57]:
```suggestion
    t.assert.strictEqual(fastify.hasRoute({
```

File: test/has-route.test.js [hunk 9]
```diff
@@ -69,7 +66,7 @@ test('hasRoute', t => {
     // parametric with regexp
     fastify.get('/example/:file(^\\d+).png', function (request, reply) { })
 
-    t.equal(fastify.hasRoute({
+    t.assert.equal(fastify.hasRoute({
```
Comment (dancastillo) [inline test/has-route.test.js:69]:
```suggestion
    t.assert.strictEqual(fastify.hasRoute({
```

File: test/has-route.test.js [hunk 10]
```diff
@@ -85,7 +82,7 @@ test('hasRoute', t => {
       }
     })
 
-    t.equal(fastify.hasRoute({
+    t.assert.equal(fastify.hasRoute({
```
Comment (dancastillo) [inline test/has-route.test.js:85]:
```suggestion
    t.assert.strictEqual(fastify.hasRoute({
```

**Review by dancastillo — COMMENTED** [review]
PR looks good, we couple spots where we should be using using `strictEqual` over `equal`

**Comment by KristianLentino99** [comment]
@dancastillo suugestions applied, very much appreciated 🙏 !

**Review by gurgunday — APPROVED** [review]
Lgtm

**Review by dancastillo — APPROVED** [review]

**Review by Eomm — APPROVED** [review]

**Pull request merged**